flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/ResourceManagerServiceImpl.java

```diff
@@ -208,6 +216,10 @@ public void revokeLeadership() {
                                 leaderSessionID);
 
                         stopLeaderResourceManager();
+
```

Comment:
Here we will trigger a `FlinkException("Unexpected termination of ResourceManagerService"` in `DispatcherResourceManagerComponent` and then force the `ClusterEntrypoint` exited. After then, Yarn and K8s will launch a new container/pod for the JobManager.